l10n_uy_ux/models/account_move.py

```diff
@@ -0,0 +1,267 @@
+            pdf_file.register_as_main_attachment(force=True)
+            self.invalidate_recordset(fnames=["invoice_pdf_report_id", "invoice_pdf_report_file"])
+        if errors := pdf_result.get("errors"):
+            msg = _("Error getting the PDF file: %s", errors)
+            self.l10n_uy_edi_error = (self.l10n_uy_edi_error or "") + msg
+            self.message_post(body=msg)
+
+    def uy_ux_action_validate_cfe(self):
+        """ Check CFE XML valid files: 350: Validación de estructura de CFE """
+        self.ensure_one()
+
+        self.l10n_uy_edi_document_id.unlink()
+        edi_doc = self.l10n_uy_edi_document_id._create_document(self)
+        self.l10n_uy_edi_document_id = edi_doc
+
+        result = edi_doc._ucfe_inbox("350", {"CfeXmlOTexto": self.l10n_uy_cfe_xml})
+        response = result.get("response")
+        if response is not None:
+            cod_rta = response.findtext(".//{*}CodRta")
+            if cod_rta != "00":
+                edi_doc._update_cfe_state(result)
+                edi_doc.message = _("Error al crear el XML del CFẸ") + "\n\n" + edi_doc.message
+                # response.Resp.CodRta  30 o 31,   01, 12, 96, 99, ? ?
+                # response.Resp.MensajeRta
+                self.l10n_uy_edi_error = _("Error al crear el XML del CFẸ\n\n %(errors)s", errors=response)
+
+    def action_l10n_uy_remkark_default(self):
+        self.ensure_one()
+        res = self.env["l10n_uy_edi.addenda"]
+
+        res |= self._uy_get_legends_recs("addenda", self)
+        res |= self._uy_get_legends_recs("cfe_doc", self)
+        res |= self._uy_get_legends_recs("emisor", self)
+        res |= self._uy_get_legends_recs("receiver", self)
+
+        for line in self._uy_get_cfe_lines():
+            res |= self._uy_get_legends_recs("item", line)
+
+        self.l10n_uy_edi_addenda_ids = res
+
+    def action_l10n_uy_addenda_preview(self):
+        self.ensure_one()
+        raise UserError(self._uy_get_cfe_addenda())
+
+    def action_l10n_uy_mandatory_legend(self):
+        self.ensure_one()
+        addenda = self._uy_get_cfe_addenda()
+        A16_InfoAdicionalDoc = self._uy_cfe_A16_InfoAdicionalDoc().get("InfoAdicionalDoc")
+        A51_InfoAdicionalEmisor = self._uy_cfe_A51_InfoAdicionalEmisor().get("InfoAdicionalEmisor")
+        A68_InfoAdicionalReceptor = self._uy_cfe_A68_InfoAdicional().get("InfoAdicional")
+        B8_DscItem = []
+        lines = self._uy_get_cfe_lines()
+        for line in lines:
+            value = self._uy_cfe_B8_DscItem(line).get("DscItem")
+            if value:
+                B8_DscItem.append((line.display_name, value))
+
+        messge = "* Adenda\n%s\n\n* Info Adicional Doc\n%s\n\n* Info Adicional Emisor\n%s\n\n* Info Adicional Receptor\n%s\n\n * Info Adicional Items\n%s" % (
+            addenda, A16_InfoAdicionalDoc, A51_InfoAdicionalEmisor, A68_InfoAdicionalReceptor, "\n".join(str(item) for item in B8_DscItem))
```

Comment:
Quizas esto tenga el mismo resultado y es mas legible! 

```suggestion
messge = (
    f"* Adenda\n{addenda}\n\n"
    f"* Info Adicional Doc\n{A16_InfoAdicionalDoc}\n\n"
    f"* Info Adicional Emisor\n{A51_InfoAdicionalEmisor}\n\n"
    f"* Info Adicional Receptor\n{A68_InfoAdicionalReceptor}\n\n"
    f"* Info Adicional Items\n{'\n'.join(str(item) for item in B8_DscItem)}"
)
```